library/bignum.c

```diff
@@ -1676,6 +1676,14 @@ int mbedtls_mpi_mul_int( mbedtls_mpi *X, const mbedtls_mpi *A, mbedtls_mpi_uint
 
     /* Calculate X*b as A + A*(b-1) to take advantage of mpi_mul_hlp */
     int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
+    /* In general, A * b requires 1 limb more than b. If
+     * A->p[n - 1] * b / b == A->p[n - 1], then A * b fits in the same
+     * number of limbs as A and the call to grow() is not required since
+     * copy() will take care of the growth. However, experimentally,
+     * making the call to grow() conditional causes slightly fewer
```

Comment:
Is this comment still correct? I'm having trouble mapping the 'turned out to be counterproductive' from the commit message with the statement that the conditional growing would cause fewer calls to `calloc`.

If the optimisation does indeed reduce the amount of `calloc`, then why wasn't it applied?

Reply:
Given the commit message and the comment, I meant to write “unconditional” rather than “conditional”. It's possible to remove the call to grow() here since the subsequent copy() grows X as needed anyway. One might think that removing grow() causes fewer allocations, since if X already has the size of A, copy() wouldn't grow it, but this call to grow() does. However, experimentally, grow() is a win because allocating the larger size now saves reallocations later.